### ISSUE TITLE
Webhooks

### ISSUE THREAD

**Issue by herder** [boxfuse/boxfuse-issues#115]
It would be nice to be able to integrate with webhooks to other systems. In our case, we would like to post a notification to Slack when an app is deployed.

**Comment by axelfontaine** [comment]
Thank you for the suggestion. Could you detail which information you feel should be present in the notification?

**Comment by herder** [comment]
This is the info we would like to use for this case:

- app name
- current app version
- app version to be deployed
- environment

and possibly:
- ami id
- AWS region

Ideally as environment variables which can be used to customize a payload, something like:

```
{
"message": "Deploying $BOXFUSE_APP:$NEW_APP_VERSION to $BOXFUSE_ENV. (Currently running version: $CURRENT_APP_VERSION"
}
```

It would be really useful to be able to set webhooks on these events:

- on deployment of a new app
- when the health of the new app is ok, and the load balancer is pointed to the new version
- when the old app is torn down

**Comment by axelfontaine** [comment]
Hmmm for Slack, wouldn't it be better to go for a [Slack Button](https://api.slack.com/docs/slack-button#) instead? The data could then be part of [attachments](https://api.slack.com/docs/message-attachments) posted in messages from incoming webhooks.

What do you think?

**Comment by herder** [comment]
Yes, that would be excellent! It's actually what we would use it for, but I didn't want to be that technology-specific... :)

**Comment by axelfontaine** [comment]
Well we actually don't mind to be technology-specific as long as the technology we integrate with is widely deployed and being specific adds clear benefits (like great usability in this case).

**Comment by herder** [comment]
Hi, just noticed the 'waiting for feedback' label on this one... :)

We're really interested in this feature, to avoid manual notifications when a deploy is starting.

**Comment by armaanbindra** [comment]
![bitmoji](https://render.bitstrips.com/v2/cpanel/8398479-171899267_15-s1-v1.png?transparent=1&palette=1&width=246)
Yes this would be really useful for us at @voiceittech as well.
Thanks,
Armaan

**Comment by khromov** [comment]
@axelfontaine Any movement on this? 👍

**Comment by axelfontaine** [comment]
@khromov Not yet.
